cirq-core/cirq/experiments/z_phase_calibration.py

```diff
@@ -271,3 +272,59 @@ def plot_z_phase_calibration_result(
         ax.set_title('-'.join(str(q) for q in pair))
         ax.legend()
     return axes
+
+
+@transformer_api.transformer
+class CalibrationTransformer:
+
+    def __init__(
+        self,
+        target: Union['cirq.Gate', 'cirq.GateFamily', 'cirq.Gateset'],
+        replacement_map: Dict[Tuple['cirq.Qid', 'cirq.Qid'], 'cirq.PhasedFSimGate'],
+    ):
+        """Create a CalibrationTransformer that replaces gates matching `target`.
+
+        Args:
+            target: The target gate, GateFamily or Gateset. Any gate matching this
+                will be replaced based on the content of `replacement_map`.
+            replacement_map:
+                A map mapping qubit pairs to calibrated gates. This is the output of
+                calling `calibrate_z_phases`.
+        """
+        self.target = ops.GateFamily(target) if isinstance(target, ops.Gate) else target
+        self.replacement_map = replacement_map
+
+    def __call__(
+        self,
+        circuit: 'cirq.AbstractCircuit',
+        *,
+        context: Optional[transformer_api.TransformerContext] = None,
+    ) -> 'cirq.Circuit':
+        """Replace every occurance of a calibrated gate with a proper replacement.
+
+        Args:
+            circuit: Circuit to transform.
+            context: Optional transformer context (not used).
+
+        Returns:
+            New circuit with each gate matching `target` and acting on a pair
+                in `replacement_map` replaced by the gate pointed to by `replacement_map`.
```

Comment:
This isn't exactly what I had in mind. The goal is to cancel the z-phases, whereas this would double them if I understand correctly.

Reply:
ptal